### PR TITLE
Improved temperature generation

### PR DESCRIPTION
**Pull request by sknot-rh**
https://github.com/ppatierno/rhte-2019/issues/15

### CODE REVIEW

File: http-device/http-device.yaml [hunk 1]
```diff
@@ -29,6 +29,10 @@ spec:
             value: device-telemetry
           - name: SEND_INTERVAL
             value: "1000"
+          - name: MIN_TEMP
+            value: "0"
+          - name: MAX_TEMP
```
Comment (ppatierno) [inline http-device/http-device.yaml]: Let's use more real values :) maybe min 20 and max 25 are an idea

File: http-device/src/main/java/io/strimzi/streams/HttpDeviceConfig.java [hunk 2]
```diff
@@ -12,17 +12,23 @@
...
     private static final String ENV_SEND_INTERVAL = "SEND_INTERVAL";
+    private static final String ENV_MIN_TEMP = "MIN_TEMP";
+    private static final String ENV_MAX_TEMP = "MAX_TEMP";
 
     private static final String DEFAULT_HOST = "localhost";
     private static final int DEFAULT_PORT = 8080;
     private static final String DEFAULT_TOPIC = "device-telemetry";
     private static final int DEFAULT_SEND_INTERVAL = 1000;
+    private static final int DEFAULT_MIN_TEMP = 0;
+    private static final int DEFAULT_MAX_TEMP = 100;
```
Comment (ppatierno) [inline http-device/src/main/java/io/strimzi/streams/HttpDeviceConfig.java]: Let's use real values as well here

File: http-device/src/main/java/io/strimzi/streams/HttpDeviceConfig.java [hunk 3]
```diff
@@ -34,12 +40,14 @@
      * @param sendInterval interval (in ms) for sending messages
      */
     private HttpDeviceConfig(String deviceId, String host, int port, 
-                             String topic, int sendInterval) {
+                             String topic, int sendInterval, int minTemp, int maxTemp) {
```
Comment (ppatierno) [inline http-device/src/main/java/io/strimzi/streams/HttpDeviceConfig.java]: Update javadoc

**Review by ppatierno — CHANGES_REQUESTED** [review]
I left a few comments, btw I would use "Temperature" instead of "Temp"

File: http-device/src/main/java/io/strimzi/streams/HttpDevice.java [hunk 4]
```diff
@@ -70,7 +70,7 @@ public void stop(Future<Void> stopFuture) throws Exception {
         Future<List<OffsetRecordSent>> fut = Future.future();        
 
         //TODO: better sensor simulation
-        int temperature = 20 + random.nextInt(5);
+        int temperature = config.getMinTemperature() + random.nextInt((config.getMaxTemperature() + 1 - config.getMinTemperature()));
```
Comment (ppatierno) [inline http-device/src/main/java/io/strimzi/streams/HttpDevice.java:72]: Why the +1. Cannot happen to have 26 as value in that case so higher than the max 25?

Comment (sknot-rh) [reply http-device/src/main/java/io/strimzi/streams/HttpDevice.java:72]: `nextInt`
Returns a random number.
between 0 (inclusive) and n (exclusive).

Comment (ppatierno) [reply http-device/src/main/java/io/strimzi/streams/HttpDevice.java:72]: Right. Si just remove the TODO comment and we are set.

**Review by ppatierno — APPROVED** [review]

**Pull request merged**